pkg/errors/errors.go

```diff
@@ -0,0 +1,158 @@
+package errors
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+
+	"github.com/marwan-at-work/vgop/semver"
+	"github.com/sirupsen/logrus"
+)
+
+// Kind enums
+const (
+	KindNotFound   = http.StatusNotFound
+	KindBadRequest = http.StatusBadRequest
+	KindUnexpected = http.StatusInternalServerError
+)
+
+// Error is an Athens system error.
+// It carries information and behavior
+// as to what caused this error so that
+// callers can implement logic around it.
+type Error struct {
+	// Kind categories Athens errors into a smaller
+	// subset of errors. This way we can generalize
+	// what an error really is: such as "not found",
+	// "bad request", etc. The official categories
+	// are HTTP status code but the ones we use are
+	// imported into this package.
+	Kind     int
```

Comment:
Could this be an enum?

Reply:
Do you mean have it as `type Kind int` and create an `iota` list? Upspin does it the way you're suggesting, however, I actually think we might hit two birds with one stone if we made the Kind be simple HTTP Statuses which could help us deduce external API errors more easily.

Reply:
Do we have to use iota? I was thinking about something like this (option without iota) https://blog.learngoprogramming.com/golang-const-type-enums-iota-bc4befd096d3

Reply:
So in this case we already have everything as an enum. See Line 12 for the enums declaration, and KindText for stringifying the enum. The only difference is that `Kind` is not its own special type. Should I do `type Kind int`?

Reply:
Bearing in mind, the user will now have to do `errors.E(op, Kind(httpStatusCode)` instead of `errors.E(op, httpStatusCode)`, or we can do the casting inside the function `case int: err.Kind = Kind(a)`

Reply:
Oh, so I can use http status and also define a custom kind? I think this is a great idea.

Reply:
Yep! I think it can play out nicely. We'll see 😄